airflow/utils/log/colored_log.py

```diff
@@ -43,6 +43,9 @@ class CustomTTYColoredFormatter(TTYColoredFormatter):
     traceback.
     """
 
+    default_time_format = '%Y-%m-%d %H:%M:%S%z'
+    default_msec_format = '%s %03dms'
```

Comment:
Why are we adding milliseconds to this PR? I don't think the logs showed ms before...

Reply:
@bbovenzi 

The original value before override in this part is `%s,%03d`.
If left as is, the file will have something like `2022-06-14 06:00:00-0600,123` which is a bit awkward. Also, it's in a format that moments.js doesn't understand.
It would be ideal to have a format like `2022-06-14 06:00:00,123-0600`, but that would be a very difficult because it would have to modify the standard library.

There are 3 ways to solve this problem:

- Avoid specifying milliseconds in the first place. by doing that we can leave the same level of information both in the UI and in the file (which is currently not. file is verbose than UI). It is possible by assigning `default_msec_format` to `None` , but this requires python3.9 or higher. Dropping 3.7 and 3.8 because of this small change ridiculous. so this is not an option. (if there is a way to consume an argument in the %-format but not print it, we can do this without dropping 3.7 and 3.8. but I don't think such a method exists.)
- In order to stick current behavior, modify the regex to match including the rear part of datetime, and then not use the rear part in replaceAll. In this case, the file side has more detailed information, anyway I think `-0600 123ms` is easier to understand than `-0600,123`.
- (which I chose) To provide the same level of information to both the UI and the file, the regex didn't match the milliseconds part. so they could exposed as they is. By doing this, UI shows more information that wasn't there before, but I didn't think that would be a problem. 

Previously, milliseconds was not shown because it was discarded from the format of moments.js added on August 3, 2018. While searching for this, I found that the most recently added format (presumably used in grid views) uses milliseconds. Based on this, I guessed that there is a demand for milliseconds.

If you think displaying milliseconds is too verbose, I'll edit the code to use the second method. However, the regular expression become a bit more complicated than it is now, and code of replaceAll become getting the milliseconds value and then ignoring it. This can be awkward code for anyone who hasn't read the conversation in this PR.

Reply:
@bbovenzi 

After I git rebase onto main:

![Cursor_및_chungking_-_Grid_-_Airflow](https://user-images.githubusercontent.com/1422445/174211685-fc527681-46f1-4102-85da-a32a68991f47.png)
> A lot of thoughts went through my mind when I saw this screen. 😢 

It would be desirable to fix it in this PR because it takes more effort to find and process it later. so I've spend some times try to fix it. it seems that if I fix the regular expression in `grid/details/content/taskInstance/Logs/utils.js` I can fix it for now, but if I do, as you concerned, we will have two log parsers.

Then I stopped trying, to ask you comment. Also, my front-end knowledge is frozen in the days of react15, so I need some time to review if I understand the grid code properly.


---

Anyway,
I wonder if logs in the grid view feature will be released in 2.3 or 2.4.
If it's going to be released in 2.4, so if this PR also has to go out in 2.4 too, I'd like to know in advance.

I'm using 1.10.15 at work, and dozens of instances are stuck with that version. I recently migrated one instance to 2.3.2 as a pilot. Then I discovered this time zone problem. Therefore, other instances are waiting for the result of this PR.

My co-worker's previous contribution was classified as a minor release, it took 5 months to release. If this PR is going to be classified as minor release, I would like to say "migrate now (with workaround config)" to my colleagues because I can guess that it will be released in a few months.